src/daemon/daemon.cpp

```diff
@@ -2023,29 +2188,44 @@ void mp::Daemon::create_vm(const CreateRequest* request, grpc::ServerWriter<Crea
 
                 reply.set_create_message("Configuring " + name);
                 server->Write(reply);
-                auto vendor_data_cloud_init_config =
-                    make_cloud_init_vendor_config(*config->ssh_key_provider, request->time_zone(), config->ssh_username,
-                                                  config->factory->get_backend_version_string().toStdString());
-                auto meta_data_cloud_init_config = make_cloud_init_meta_config(name);
-                auto user_data_cloud_init_config = YAML::Load(request->cloud_init_user_data());
-                prepare_user_data(user_data_cloud_init_config, vendor_data_cloud_init_config);
 
-                std::string mac_addr;
-                while (true)
+                // Check that we have at least one network interface.
+                if (0 == checked_args.interfaces.size())
                 {
-                    mac_addr = mp::utils::generate_mac_address();
+                    throw std::runtime_error("No network interfaces");
+                }
+
+                // Generate MAC addresses for the interfaces on which it was not specified and put the modified
+                // interfaces in a new vector. Create another vector containing only the interfaces which must be
+                // configured via cloud-init.
+                std::vector<mp::NetworkInterface> interfaces, cloud_init_interfaces;
 
-                    auto it = allocated_mac_addrs.find(mac_addr);
-                    if (it == allocated_mac_addrs.end())
+                for (auto iface : checked_args.interfaces)
+                {
+                    if ("" == iface.first.mac_address)
```

Comment:
```suggestion
                    if (iface.first.mac_address.empty())
```

Reply:
I separated the default network interface from the list of extra interfaces. This is resolved.